service/src/java/org/apache/hive/service/cli/thrift/ThriftBinaryCLIService.java

```diff
@@ -105,4 +140,79 @@ public void run() {
     }
   }
 
+  // Custom authentication class with Kerberos thread
+  private static void startCustomWithKerberos(
+   final HiveConf hiveConf,
+   final ThriftCLIService service,
+   final String hiveHost,
+   TServer customKrbServer) throws Exception {
```

Comment:
this is tricky... you mask the instance variable, customKrbServer, then pass it in when you call it.  So it all works, but is definitely unclear.  just remove the argument and reference the member directly.